source/guide_synapse.rst

```diff
@@ -0,0 +1,412 @@
+    Installing collected packages: MarkupSafe, jinja2
+    Successfully installed MarkupSafe-1.1.1 jinja2-2.10.1
+  (env) [isabell@stardust ~]$ pip install psycopg2
+    Collecting psycopg2
+      Downloading https://files.pythonhosted.org/packages/5c/1c/6997288da181277a0c29bc39a5f9143ff20b8c99f2a7d059cfb55163e165/psycopg2-2.8.3.tar.gz (377kB)
+         |████████████████████████████████| 378kB 21.4MB/s
+    Building wheels for collected packages: psycopg2
+      Building wheel for psycopg2 (setup.py) ... done
+      Stored in directory: /home/matrites/.cache/pip/wheels/48/06/67/475967017d99b988421b87bf7ee5fad0dad789dc349561786b
+    Successfully built psycopg2
+    Installing collected packages: psycopg2
+    Successfully installed psycopg2-2.8.3
+
+  (env) [isabell@stardust ~]$
+
+
+Configuration
+=============
+
+Generate standard config
+------------------------
+
+Generate a config file ``~/synapse/homeserver.yaml`` and replace my.domain.name with the real domain you want your matrix username to end in.
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ source ~/synapse/env/bin/activate
+  (env) [isabell@stardust ~]$ cd ~/synapse
+  (env) [isabell@stardust synapse]$  python -m synapse.app.homeserver \
+    --server-name my.domain.name \
+    --config-path homeserver.yaml \
+    --generate-config \
+    --report-stats=[yes|no]
+  (env) [isabell@stardust ~]$
+
+Set the Synapse_ to listen for federation and clients on port 8008 without encryption in the config file ``~/synapse/homeserver.yaml``.
+
+.. code-block:: yaml
+
+      - port: 8008
+        type: http
+        tls: false
+        bind_addresses: ['::','0.0.0.0']
+        x_forwarded: true
+        resources:
+          - names: [client, federation]
+
+And point the ``uberspace web backend`` on ``/`` to the listener on port 8008.
+
+.. include:: includes/web-backend.rst
+
+To enable federation as described MatrixFederation_ we need to announce, that we are listening on port 443 (the reverse proxy), either via DNS or via .well-known.
+
+DNS announcement
+----------------
+
+The port can can be announced by setting a DNS record in the following format:
+
+.. code-block::
+
+   _matrix._tcp.<yourdomain.com> <ttl> IN SRV <priority> <weight> <port> <synapse.server.name>
+
+For example like this:
+
+.. code-block::
+
+  _matrix._tcp.my.domain.name 3600 IN SRV 10 5 443 my.domain.name.
+
+.. note:: this can be checked by running ``dig -t srv _matrix._tcp.my.domain.name``
+
+.well-known announcement
+------------------------
+
+The federation port can also be announced via a file ``~/html/.well-known/matrix/server``
+
+.. code-block:: json
+  {
+    "m.server": "my.domain.name:443"
+  }
+
+This has to be made available under ``/.well-known/matrix`` via the web backend:
+
+.. include:: includes/web-backend.rst
+
+Configure Certificates
```

Comment:
`server_name` needs to be updated to match the hostname and certificate. Otherwise the federation checker fails with error `MatchingServerName`. Can you please add it? :)

Reply:
Oh, my bad. It does!